airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -496,17 +517,15 @@ def start_java_dataflow(
         variables['jobName'] = name
         variables['region'] = location
 
-        def label_formatter(labels_dict):
-            return ['--labels={}'.format(
-                json.dumps(labels_dict).replace(' ', ''))]
+        if 'labels' in variables:
+            variables['labels'] = json.dumps(variables['labels']).replace(' ', '')
```

Comment:
This is modifying user provided input. Is this your intention?

Reply:
might be better to check that labels adhere to these regex (in API [docs](https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#Job)):
```
Keys must conform to regexp: [\p{Ll}\p{Lo}][\p{Ll}\p{Lo}\p{N}_-]{0,62}
Values must conform to regexp: [\p{Ll}\p{Lo}\p{N}_-]{0,63}
```
and raise exception.

Reply:
@jaketf do you know maybe are these regexes are used for labels by all google apis?
I see that google provides a comprehensive information when labels do not comply with these regex, with regex included, however we could make validation of labels for whole google provider and throw some warning to the user (before it fails during run).
WDYT @jaketf @mik-laj

Reply:
On the other hand, I don't think it is good idea to add unnecessary complexity to the  it and limit user. Google logs already provide extensive information about problems with labels when they occur

Reply:
> This is modifying user provided input. Is this your intention?

@aaltay the reason of this replace was to avoid spaces in json, change of the user input was side effect.
I changed it to `variables['labels'] = json.dumps(variables['labels'], separators=(',', ':'))` so the json is compact as well and labels provided by the user are not touched.